File: 476. Number Complement/bitwise.py
# Python 3

class Solution:
  def findComplement(self, num : int) -> int:
    self.x = num
    num |= num >> 1
    num |= num >> 2
    num |= num >> 4
    num |= num >> 8
    num |= num >> 16
    return self.x ^ num

# Python 2

class Solution(object):
  def findComplement(self, num):
    """
    :type num: int
    :rtype: int
    """
    self.x = num
    num |= num >> 1
    num |= num >> 2
    num |= num >> 4
    num |= num >> 8
    num |= num >> 16
    return self.x ^ num

File: 476. Number Complement/bitwise.cpp
class Solution {
public:
  int pollute(int number){
    number |= number >> 1;
    number |= number >> 2;
    number |= number >> 4;
    number |= number >> 8;
    number |= number >> 16;
    return number;
  }
  int finComplement(int num){
    return num ^ pollute(num);
  }
};
